### PR TITLE
Paper - Katrina K - C15

### PR DESCRIPTION
**Pull request by SterlingSunshine**

### CODE REVIEW

File: swap_meet/clothing.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Swap Meet
+# Ada Cohort 15 - Paper
+# Katrina Kimzey (she|they)
+# Started April 3, 2021
+
+from .item import Item
+
+class Clothing(Item):
```
Comment (CheezItMan) [inline swap_meet/clothing.py:8]: 👍 Very clean child classes!

File: swap_meet/item.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Swap Meet
+# Ada Cohort 15 - Paper
+# Katrina Kimzey (she|they)
+# Started March 31, 2021
+
+class Item:
```
Comment (CheezItMan) [inline swap_meet/item.py:6]: 👍

File: swap_meet/vendor.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# Swap Meet
+# Ada Cohort 15 - Paper
+# Katrina Kimzey (she|they)
+# Started March 31, 2021
+
+class Vendor:
+    def __init__(self, inventory = None):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:7]: 👍

File: swap_meet/vendor.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        # include an inventory
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+        
+
+    def add(self, item):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:15]: 👍

File: swap_meet/vendor.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        # inventory entry
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:20]: 👍 , nice use of try-except.

File: swap_meet/vendor.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        try:
+            self.inventory.remove(item)
+        except ValueError as err:
+            return False
+
+        return item
+
+    def get_by_category(self, category):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:28]: 👍

File: swap_meet/vendor.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    def swap_items(self, friend, my_item, their_item):
+        if (my_item in self.inventory and 
+            their_item in friend.inventory):
+            self.remove(my_item)
+            friend.remove(their_item)
+            self.add(their_item)
+            friend.add(my_item)
+            return True
+        else:
+            return False
```
Comment (CheezItMan) [inline swap_meet/vendor.py:46]: The `else` is not needed.  We have a number of these in `Vendor`

```suggestion
        return False
```

File: swap_meet/vendor.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        
+    def swap_first_item(self, friend):
+        if len(self.inventory) and len(friend.inventory):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:49]: You don't actually need the `len`

```suggestion
        if self.inventory and friend.inventory:
```

File: swap_meet/vendor.py [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+            their_item in friend.inventory):
+            self.remove(my_item)
+            friend.remove(their_item)
+            self.add(their_item)
+            friend.add(my_item)
+            return True
+        else:
+            return False
+        
+    def swap_first_item(self, friend):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:48]: 👍

File: swap_meet/vendor.py [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        if len(self.inventory) and len(friend.inventory):
+            self.swap_items(friend, self.inventory[0], friend.inventory[0])
+            return True
+        else:
+            return False
+
+    def get_best_by_category(self, category):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:55]: 👍

File: swap_meet/vendor.py [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+        
+        best_item = items_of_category[0]
+
+        for item in items_of_category:
+            if best_item.condition < item.condition:
+                best_item = item
+
+        return best_item
+
+    def swap_best_by_category(self, other, my_priority, their_priority):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:69]: 👍 Good use of prior methods

File: swap_meet/vendor.py [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+        mine_to_swap = self.get_best_by_category(their_priority)
+        theirs_to_swap = other.get_best_by_category(my_priority)
+
+        if mine_to_swap and theirs_to_swap:
+            self.swap_items(other, mine_to_swap, theirs_to_swap)
+            return True
+        else:
+            return False
+
+    def swap_by_newest(self, other):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:79]: 👍 Nice work

File: tests/test_wave_07.py [hunk 13]
```diff
@@ -0,0 +1,33 @@
+import pytest
+from swap_meet.vendor import Vendor
+from swap_meet.item import Item
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
+def test_swap_by_age():
```
Comment (CheezItMan) [inline tests/test_wave_07.py:8]: Nice that you got started testing here.  You should also include edge-case tests with empty lists of inventory and where the ages are tied.

**Review by CheezItMan — COMMENTED** [review]
Nice work Katrina, you hit the learning goals here.  Well done.  I made a few suggestions, mostly on coding style, but this was very well done.
